subscription-service/src/main/kotlin/com/egm/stellio/subscription/web/SubscriptionHandler.kt

```diff
@@ -39,15 +39,12 @@ class SubscriptionHandler(
                     Pair(parseSubscription(it.t1, context), it.t2.subject)
                 }
                 .flatMap { subscriptionAndSubject ->
-                    subscriptionService.exists(subscriptionAndSubject.first.id).flatMap {
-                        Mono.just(Triple(subscriptionAndSubject.first, it, subscriptionAndSubject.second))
+                    isNewSubscription(subscriptionAndSubject.first.id).flatMap {
```

Comment:
split it into two `flatMap`, it is more readable

Reply:
Hmm, the inner `flatMap` even seems useless

Reply:
i need to call flatMap on isNewSubscription in order to consume its value.
i used nested flatMaps to be able to pass the payload for further treatement